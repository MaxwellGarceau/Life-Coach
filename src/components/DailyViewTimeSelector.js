import React from 'react';
import { connect } from 'react-redux';

// JSX for selector
// let timeOptions = '';
// for (let i = 0; i < 12; i++) {
//   timeOptions += <option>{i + 1}am</option>;
// };
// for (let i = 0; i < 12; i++) {
//   timeOptions += <option>{i + 1}pm</option>;
// };
// timeOptions = (
//   <select>{timeOptions}</select>
// );
// console.log(timeOptions);

const DailyViewTimeSelector = () => {
  return (
    <div>
      <select>
        <option>12am</option>
        <option>1am</option>
        <option>2am</option>
        <option>3am</option>
        <option>4am</option>
        <option>5am</option>
        <option>6am</option>
        <option>7am</option>
        <option>8am</option>
        <option>9am</option>
        <option>10am</option>
        <option>11am</option>
        <option>12pm</option>
        <option>1pm</option>
        <option>2pm</option>
        <option>3pm</option>
        <option>4pm</option>
        <option>5pm</option>
        <option>6pm</option>
        <option>7pm</option>
        <option>8pm</option>
        <option>9pm</option>
        <option>10pm</option>
        <option>11pm</option>
      </select>
      <span>-</span>
      <select>
        <option>12am</option>
        <option>1am</option>
        <option>2am</option>
        <option>3am</option>
        <option>4am</option>
        <option>5am</option>
        <option>6am</option>
        <option>7am</option>
        <option>8am</option>
        <option>9am</option>
        <option>10am</option>
        <option>11am</option>
        <option>12pm</option>
        <option>1pm</option>
        <option>2pm</option>
        <option>3pm</option>
        <option>4pm</option>
        <option>5pm</option>
        <option>6pm</option>
        <option>7pm</option>
        <option>8pm</option>
        <option>9pm</option>
        <option>10pm</option>
        <option>11pm</option>
      </select>
    </div>
  )
};

export default DailyViewTimeSelector;
