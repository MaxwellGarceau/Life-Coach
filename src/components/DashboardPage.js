import React from 'react';
import LifeGoals from './life-goals/LifeGoals';

const DashboardPage = () => (
  <div>
    <LifeGoals />
  </div>
);

export default DashboardPage;
