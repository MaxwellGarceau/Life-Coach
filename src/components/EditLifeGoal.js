import React from 'react';
import { connect } from 'react-redux';
import { startEditLifeGoal, startRemoveLifeGoal } from '../actions/life-goals';
import { startRemoveNote } from '../actions/notes';
import determineDeleteGoalNotes from '../selectors/determine-delete-goal-notes';
import LifeGoalForm from './LifeGoalForm';

class EditLifeGoal extends React.Component {
  onSubmit = (lifeGoal) => {
    this.props.startEditLifeGoal(this.props.goal.id, lifeGoal);
    this.props.history.push("/goals");
  }
  handleOnRemove = () => {
    let confirmRemoveGoal = confirm('Warning: Your calendar events assigned to this life goal will also be deleted. If you wish to change your life goal simply input your new life goal and click accept. Are you sure you wish to continue?');
    if (!confirmRemoveGoal) {
      return;
    }
    const deleteNoteArr = determineDeleteGoalNotes(this.props.notes, this.props.goal.id);
    deleteNoteArr.map((note) => this.props.startRemoveNote(note.id));
    this.props.startRemoveLifeGoal({ id: this.props.goal.id });
    this.props.history.push("/goals");
  }
  render (props) {
    return (
      <section>
        <h1>Edit Life Goal</h1>
        <div>
        <LifeGoalForm
          goal={this.props.goal}
          onSubmit={this.onSubmit}
        />
        <button className="button button--secondary" onClick={this.handleOnRemove}>Remove Goal</button>
        </div>
      </section>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startEditLifeGoal: (id, lifeGoal) => dispatch(startEditLifeGoal(id, lifeGoal)),
  startRemoveLifeGoal: (id) => dispatch(startRemoveLifeGoal(id)),
  startRemoveNote: (note) => dispatch(startRemoveNote(note))
})

const mapStateToProps = (state, props) => ({
  goal: state.lifeGoals.find((goal) => goal.id === props.match.params.id),
  notes: state.notes
});

export default connect(mapStateToProps, mapDispatchToProps)(EditLifeGoal);
