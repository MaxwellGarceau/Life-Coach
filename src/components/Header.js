import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/dashboard">
          <h1>Life Coach</h1>
        </Link>
        <Link to="/goals">
          <button className="button button--link">Life Goals</button>
        </Link>
        <Link to="/daily-calendar">
          <button className="button button--link">Daily Calendar</button>
        </Link>
        <Link to="/calendar">
          <button className="button button--link">Calendar</button>
        </Link>
        <button className="button button--link" onClick={startLogout}>Logout</button>
      </div>
    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
