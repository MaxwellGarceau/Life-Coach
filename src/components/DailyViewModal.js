import React from 'react';
import { connect } from 'react-redux';
import DailyViewTimeSelector from './DailyViewTimeSelector';

export class DailyViewModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      noteDescription: props.activity ? props.activity.noteDescription : '',
      currentStartTime: this.props.defaultStartTime,
      currentEndTime: this.defaultEndTime()
    };
  }
  defaultEndTime = () => {
    let endTime = this.props.defaultStartTime;
    let endTimeNum = endTime.match(/\d+/g);
    endTimeNum = parseInt(endTimeNum, 10) + 2;
    endTime = endTime.replace(/\d+/g, String(endTimeNum));
    switch (endTime) {
      case '13am':
        endTime = '1pm';
        break;
      case '14am':
        endTime = '2pm';
        break;
      case '13pm':
        endTime = '1am';
        break;
      case '14pm':
        endTime = '2am';
        break;
    }
    return endTime;
  };
  onNoteDescriptionChange = (e) => {
    const noteDescription = e.target.value;
    this.setState(() => ({ noteDescription }));
  };
  onSubmit = (e) => {
    e.preventDefault();
    // Send currentStartTime and currentEndTime to redux
    // Send note Description to redux instead
    this.props.setNoteDescription(this.state.noteDescription);
    this.onToggleModal();
  };
  onToggleModal = () => {
    this.props.onToggleModal();
  };
  onClearNoteDescription = () => {
    this.props.setNoteDescription('');
    this.onToggleModal();
  };
  handleStartTimeOnChange = (currentStartTime) => {
    this.setState({ currentStartTime });
  };
  handleEndTimeOnChange = (currentEndTime) => {
    this.setState({ currentEndTime });
  };
  render() {
    return (
      <div className="daily-view-modal">
        <form id="daily-view-form" onSubmit={this.onSubmit}>
          <h6 className="daily-view-modal__title">Planner Modal</h6>
          <textarea
            className="daily-view-modal__body"
            onChange={this.onNoteDescriptionChange}
            value={this.state.noteDescription}
            placeholder="Activity Description"
          />
          <DailyViewTimeSelector
            defaultStartTime={this.props.defaultStartTime}
            defaultEndTime={this.defaultEndTime()}
            handleStartTimeOnChange={this.handleStartTimeOnChange}
            handleEndTimeOnChange={this.handleEndTimeOnChange}
          />
          <div>
            <button className="button daily-view-modal__button" type="submit">
              Save
            </button>
            <button
              className="button daily-view-modal__button"
              type="button"
              onClick={this.onClearNoteDescription}
            >
              Clear Contents
            </button>
            <button
              className="button daily-view-modal__button"
              type="button"
              onClick={this.onToggleModal}
            >
              Cancel
            </button>
          </div>
        </form>
      </div>
    );
  }
}

export default DailyViewModal;
