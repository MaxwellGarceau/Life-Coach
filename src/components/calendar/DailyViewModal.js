import React from 'react';
import { connect } from 'react-redux';
import { startAddNote } from '../../actions/notes';
import DailyViewTimeSelector from './DailyViewTimeSelector';
import ModalGoalSelection from './ModalGoalSelection';
import determineElapsedTime, { defaultEndTime } from '../../selectors/determine-elapsed-time';

export class DailyViewModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      noteDescription: props.assignedNote ? props.assignedNote.noteDescription : '',
      currentStartTime: props.assignedNote ? props.assignedNote.currentStartTime : this.props.defaultStartTime,
      currentEndTime: props.assignedNote ? props.assignedNote.currentEndTime : defaultEndTime(this.props.defaultStartTime)
    };
  }
  onNoteDescriptionChange = (e) => {
    const noteDescription = e.target.value;
    this.setState(() => ({ noteDescription }));
  };
  onSubmit = (e) => {
      e.preventDefault();
    // Maybe also send an object with an array of times in between to redux
    const note = {
      noteDescription: this.state.noteDescription,
      currentStartTime: this.state.currentStartTime,
      currentEndTime: this.state.currentEndTime,
      elapsedTime: determineElapsedTime(this.state.currentStartTime, this.state.currentEndTime),
      goalId: this.state.goalId,
      currentDate: this.props.currentDate
    };
    this.props.startAddNote(note);
    this.onToggleModal();
  };
  onToggleModal = () => {
    this.props.onToggleModal();
  };
  // Need to delete note from redux with this function
  onClearNoteDescription = () => {
      this.onToggleModal();
  };
  handleStartTimeOnChange = (currentStartTime) => {
    this.setState({ currentStartTime });
  };
  handleEndTimeOnChange = (currentEndTime) => {
    this.setState({ currentEndTime });
  };
  handleGoalId = (goalId) => {
    this.setState({ goalId });
  }
  render() {
    return (
      <div className="daily-view-modal">
        <form id="daily-view-form" onSubmit={this.onSubmit}>
          <h6 className="daily-view-modal__title">Planner Modal</h6>
          <textarea
            className="daily-view-modal__body"
            onChange={this.onNoteDescriptionChange}
            value={this.state.noteDescription}
            placeholder="Activity Description"
          />
          <DailyViewTimeSelector
            defaultStartTime={this.props.assignedNote ? this.props.assignedNote.currentStartTime : this.props.defaultStartTime}
            defaultEndTime={this.props.assignedNote ? this.props.assignedNote.currentEndTime : ''}
            handleStartTimeOnChange={this.handleStartTimeOnChange}
            handleEndTimeOnChange={this.handleEndTimeOnChange}
          />
          <ModalGoalSelection 
            handleGoalId={this.handleGoalId}
            assignedNote={this.props.assignedNote}
          />
          <div>
            <button className="button daily-view-modal__button" type="submit">
              Save
            </button>
            <button
              className="button daily-view-modal__button"
              type="button"
              onClick={this.onClearNoteDescription}
            >
              Clear Contents
            </button>
            <button
              className="button daily-view-modal__button"
              type="button"
              onClick={this.onToggleModal}
            >
              Cancel
            </button>
          </div>
        </form>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddNote: (note) => dispatch(startAddNote(note))
});

const mapStateToProps = (state, ownProps) => ({
  // Using props that are passed down instead of redux for currentDate
  // currentDate: state.calendar.currentDate
});

export default connect(mapStateToProps, mapDispatchToProps)(DailyViewModal);
