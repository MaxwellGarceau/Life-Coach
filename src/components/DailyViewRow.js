// Need to save note to redux so I can edit it and access it later
import React from 'react';
import { connect } from 'react-redux';
import DailyViewModal from './DailyViewModal';
let temp = false;

export class DailyViewRow extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isModalVisible: false
    };
  }
  onToggleModal = () => {
    (this.state.isModalVisible == false) ? this.setState(() => ({ isModalVisible: true })) : this.setState(() => ({ isModalVisible: false }));
  };
  setNoteDescription = (noteDescription) => {
    this.setState(() => ({ noteDescription }));
  };
  render (props) {
    let modalBgVisiblityClasses = this.state.isModalVisible ? ' visible opacity-full' : '';
    return (
      <section className={`calendar__row--bg-color${(this.props.test == this.props.defaultStartTime) ? ' activity-bg-color' : ''}`}>
      {console.log(this.props.test)}
    {/* Individual Calendar Row */}
        <div id={this.props.defaultStartTime} className="calendar__row" onClick={this.onToggleModal}>
          <div>{this.props.defaultStartTime}</div>
          <div className="daily-view__description">{this.state.noteDescription}</div>
        </div>
      {/* Modal pop up for inputting activity */}
        <div className={`daily-view-modal__background${modalBgVisiblityClasses}`}>{this.state.isModalVisible && <DailyViewModal setNoteDescription={this.setNoteDescription} onToggleModal={this.onToggleModal} defaultStartTime={this.props.defaultStartTime}/>}</div>
      </section>
    );
  }
}

// const mapDispatchToProps = dispatch => ({
//   handleCreateActivity: expense => dispatch(temp)
// });

const mapStateToProps = (state) => ({
  // Replace with actual item from redux store later
  test: !!state.notes[0] ? state.notes[0].currentStartTime : ''
});

export default connect(mapStateToProps)(DailyViewRow);
