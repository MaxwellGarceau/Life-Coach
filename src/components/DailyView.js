import React from 'react';
import { DailyViewRow } from './DailyViewRow';

export class DailyView extends React.Component {
  render () {
    return (
      <div className="calendar">
        <DailyViewRow time={'12am'} />
        <DailyViewRow time={'1am'} />
        <DailyViewRow time={'2am'} />
        <DailyViewRow time={'3am'} />
        <DailyViewRow time={'4am'} />
        <DailyViewRow time={'5am'} />
        <DailyViewRow time={'6am'} />
        <DailyViewRow time={'7am'} />
        <DailyViewRow time={'8am'} />
        <DailyViewRow time={'9am'} />
        <DailyViewRow time={'10am'} />
        <DailyViewRow time={'11am'} />
        <DailyViewRow time={'12pm'} />
        <DailyViewRow time={'1pm'} />
        <DailyViewRow time={'2pm'} />
        <DailyViewRow time={'3pm'} />
        <DailyViewRow time={'4pm'} />
        <DailyViewRow time={'5pm'} />
        <DailyViewRow time={'6pm'} />
        <DailyViewRow time={'7pm'} />
        <DailyViewRow time={'8pm'} />
        <DailyViewRow time={'9pm'} />
        <DailyViewRow time={'10pm'} />
        <DailyViewRow time={'11pm'} />
      </div>
    );
  }
}
