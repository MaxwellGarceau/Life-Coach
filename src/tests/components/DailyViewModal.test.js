import React from 'react';
import { shallow } from 'enzyme';
import { DailyViewModal } from '../../components/DailyViewModal';

const defaultStartTime = '1am';

test('should render DailyViewModal correctly', () => {
  const wrapper = shallow(<DailyViewModal defaultStartTime={defaultStartTime} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render note description change', () => {
  const value = 'My Note';
  const wrapper = shallow(<DailyViewModal defaultStartTime={defaultStartTime} />);
  wrapper.find('textarea').simulate('change', {
    target: { value }
  });
  expect(wrapper.state('noteDescription')).toBe(value);
});

test('should call setNoteDescription prop for form submission', () => {
  const setNoteDescriptionSpy = jest.fn();
  const onToggleModalSpy = jest.fn();
  const startAddNote = jest.fn();
  const value = 'My Note';  
  const wrapper = shallow(<DailyViewModal startAddNote={startAddNote} defaultStartTime={defaultStartTime} setNoteDescription={setNoteDescriptionSpy} onToggleModal={onToggleModalSpy}/>);
  wrapper.find('textarea').simulate('change', {
    target: { value }
  });
  wrapper.find('form').simulate('submit', {
    preventDefault: () => {}
  });
  expect(setNoteDescriptionSpy).toHaveBeenLastCalledWith(value);
});

test('should call onToggleModal prop for form submission', () => {
  const setNoteDescriptionSpy = jest.fn();
  const onToggleModalSpy = jest.fn();
  const startAddNote = jest.fn();
  const wrapper = shallow(<DailyViewModal startAddNote={startAddNote} defaultStartTime={defaultStartTime} setNoteDescription={setNoteDescriptionSpy} onToggleModal={onToggleModalSpy}/>);
  wrapper.find('form').simulate('submit', {
    preventDefault: () => {}
  });
  expect(onToggleModalSpy).toHaveBeenCalled();
});

test('should call onToggleModal prop on cancel button click', () => {
  const onToggleModalSpy = jest.fn();
  const wrapper = shallow(<DailyViewModal defaultStartTime={defaultStartTime} onToggleModal={onToggleModalSpy}/>);
  wrapper.find('button').at(2).simulate('click');
  expect(onToggleModalSpy).toHaveBeenCalled();
});

test('should clear note description on clear contents button click', () => {
  const setNoteDescriptionSpy = jest.fn();
  const onToggleModalSpy = jest.fn();
  const wrapper = shallow(<DailyViewModal defaultStartTime={defaultStartTime} setNoteDescription={setNoteDescriptionSpy} onToggleModal={onToggleModalSpy}/>);
  wrapper.find('button').at(1).simulate('click');
  expect(wrapper.state('noteDescription')).toBe('');
});

test('should call onToggleModal prop on clear contents button click', () => {
  const setNoteDescriptionSpy = jest.fn();
  const onToggleModalSpy = jest.fn();
  const wrapper = shallow(<DailyViewModal defaultStartTime={defaultStartTime} setNoteDescription={setNoteDescriptionSpy} onToggleModal={onToggleModalSpy}/>);
  wrapper.find('button').at(1).simulate('click');
  expect(onToggleModalSpy).toHaveBeenCalled();
});

// Create test for redux (start add note was called)
