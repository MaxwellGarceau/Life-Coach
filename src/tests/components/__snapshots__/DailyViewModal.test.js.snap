// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DailyViewModal correctly 1`] = `
<div
  className="daily-view-modal"
>
  <form
    id="daily-view-form"
    onSubmit={[Function]}
  >
    <h6
      className="daily-view-modal__title"
    >
      Planner Modal
    </h6>
    <textarea
      className="daily-view-modal__body"
      onChange={[Function]}
      placeholder="Activity Description"
      value=""
    />
    <DailyViewTimeSelector
      defaultEndTime="3am"
      defaultStartTime="1am"
      handleEndTimeOnChange={[Function]}
      handleStartTimeOnChange={[Function]}
    />
    <div>
      <button
        className="button daily-view-modal__button"
        type="submit"
      >
        Save
      </button>
      <button
        className="button daily-view-modal__button"
        onClick={[Function]}
        type="button"
      >
        Clear Contents
      </button>
      <button
        className="button daily-view-modal__button"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</div>
`;
